pipeline_steps/run_tests.groovy

```diff
@@ -0,0 +1,71 @@
+def run_irr_tests() {
+  common.conditionalStage(
+    stage_name: "Run IRR Tests",
+    stage: {
+      pubcloud.runonpubcloud {
+        try {
+          ansiColor('xterm') {
+            dir("${env.WORKSPACE}/${env.ghprbGhRepository}") {
+              print("Triggered by PR: ${env.ghprbPullLink}")
+              checkout([$class: 'GitSCM',
+                branches: [[name: "origin/pr/${env.ghprbPullId}/merge"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[$class: 'CleanCheckout']],
+                submoduleCfg: [],
+                userRemoteConfigs: [
+                  [
+                    url: "https://github.com/${env.ghprbGhRepository}.git",
+                    refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
+                  ]
+                ]
+              ]) // checkout
+              sh """#!/bin/bash
+              bash ./run_tests.sh
+              """
+            } // dir
+          } // ansiColor
+        } catch (e) {
+          print(e)
+          throw e
+        } finally {
+          irr_archive_artifacts()
+        }
+      } // pubcloud slave
+    } // stage
+  ) // conditionalStage
+}
+
+def irr_archive_artifacts(){
+  dir("${env.WORKSPACE}/logs") {
+    print("Running Log Collection")
+    try{
+      sh """#!/bin/bash
+      d="artifacts_\${BUILD_TAG}"
+      touch "\$d".marker
+      tar -C "${WORKSPACE}/logs" -cjf "${env.WORKSPACE}/\$d".tar.bz2 .
```

Comment:
Will this try to tarball itself? Maybe add an exclude for the file being created.